slick/src/main/scala/slick/jdbc/H2Profile.scala

```diff
@@ -125,6 +126,11 @@ trait H2Profile extends JdbcProfile {
     }
   }
 
+  class TableDDLBuilder(table: Table[_]) extends super.TableDDLBuilder(table) {
+    // primary key will be dropped when the table is dropped
+    override protected def dropIfExistsPhase = Iterable(dropTable(true))
```

Comment:
It has to, otherwise the index would refer to nonexistent objects.

Can you explain why you're overriding this? What does the superclass do?

Reply:
https://github.com/slick/slick/pull/2228/files#diff-d943dbb8f0a265a24f785cae34bcf1c36840719fcfab398dd7fe84c91f25acd5R310

I remember that the decision was made that drop table was unnecessary because it deletes the index, just like Postgres does.
I think I read the documentation or actually got it to work. Please wait a bit as I look for the reasoning behind it. (I can't take the time right now.)

Reply:
> What does the superclass do?

```scala
    protected def dropIfExistsPhase = primaryKeys.map(dropPrimaryKey) ++Iterable(dropTable(true))
```
In the supuer class, after dropping the primary key index, DROP TABLE is executed.
I think this is because in some products, it is necessary to consider table deletion and primary key deletion separately, but in H2 Database, since the primary key index is also deleted when the table is deleted, I think there is no problem with just the DROP TABLE statement.

I actually tested this by issuing the following SQL.

```sql

CREATE TABLE FOO(ID PRIMARY KEY, VALUE INT);
INSERT INTO FOO VALUES (1,1);
INSERT INTO FOO VALUES (2,2);
INSERT INTO FOO VALUES (2,2);
SELECT * FROM INFORMATION_SCHEMA.INDEXES ;
DROP TABLE FOO;
SELECT * FROM INFORMATION_SCHEMA.INDEXES ;
```